tests/unit/input_time_test.py

```diff
@@ -0,0 +1,197 @@
+    assert input.widget.v_model == dt.time(10, 17).strftime("%H:%M")
+
+
+def test_input_time_incomplete_entry():
+    on_value = MagicMock()
+    on_v_model = MagicMock()
+
+    @solara.component
+    def Test():
+        def update_value(value: dt.time):
+            on_value(value)
+            set_value(value)
+
+        value, set_value = solara.use_state(now)
+
+        InputTime(value=value, on_value=update_value, label="label")
+
+    el = Test()
+    box, rc = solara.render(el, handle_error=False)
+    input = rc.find(vw.TextField)
+    input.widget.observe(on_v_model, "v_model")
+
+    input.widget.v_model = "10"
+    assert on_value.call_count == 0
+    assert on_v_model.call_count == 1
+    assert on_value.call_args is None
+    assert input.widget.error
+    assert input.widget.v_model == "10"
+    assert on_v_model.call_count == 1
+
+    input.widget.v_model = "10:17"
+    assert on_value.call_count == 1
+    assert on_v_model.call_count == 2
+    assert on_value.call_args is not None
+    assert on_value.call_args[0][0] == dt.time(10, 17)
+    assert not input.widget.error
+    assert input.widget.v_model == dt.time(10, 17).strftime("%H:%M")
+
+
+def test_input_time_invalid_format():
+    on_value = MagicMock()
+    on_v_model = MagicMock()
+
+    @solara.component
+    def Test():
+        def update_value(value: dt.time):
+            on_value(value)
+            set_value(value)
+
+        value, set_value = solara.use_state(now)
+
+        InputTime(value=value, on_value=update_value, label="label", time_format="%I:%M %p")
+
+    el = Test()
+    box, rc = solara.render(el, handle_error=False)
+    input = rc.find(vw.TextField)
+    input.widget.observe(on_v_model, "v_model")
+
+    input.widget.v_model = "2:30 PM"
+    assert on_value.call_count == 1
+    assert on_v_model.call_count == 1
+    assert on_value.call_args is not None
+    assert on_value.call_args[0][0] == dt.time(14, 30)
+    assert not input.widget.error
+
+    input.widget.v_model = "25:00 PM"
+    assert on_value.call_count == 1
+    assert on_v_model.call_count == 2  # This is still called to reflect change
+    assert on_value.call_args[0][0] == dt.time(14, 30)  # Value remains unchanged due to invalid input
+    assert input.widget.error
+    assert input.widget.label == "label (Time 25:00 PM does not match format %I:%M %p)"
+
+
+def test_input_time_on_open_value():
+    on_open_value = MagicMock()
+    on_value = MagicMock()
+    el = InputTime(value=now, label="label", on_value=on_value, on_open_value=on_open_value)
+    box, rc = solara.render(el, handle_error=False)
+    menu = rc.find(vw.Menu)
+    assert menu is not None
+
+    # Simulate opening the time picker
+    menu.widget.v_model = True
+    assert on_open_value.call_count == 1
+    assert on_open_value.call_args[0][0] is True
+
+    # Simulate closing the time picker
+    menu.widget.v_model = False
+    assert on_open_value.call_count == 2
+    assert on_open_value.call_args[0][0] is False
```

Comment:
Now that I tried it myself, I realized that since the menu is front-end only (Vuetify template), this can't be done in a unit test. We could test this in the integration tests, but we effectively already do since `Menu` has its own tests.